satpy/composites/__init__.py

```diff
@@ -700,12 +703,9 @@ def _init_reflectance_calculator(self, metadata):
             LOG.info("Couldn't load pyspectral")
             raise ImportError("No module named pyspectral.near_infrared_reflectance")
 
-        if self.sun_zenith_threshold is not None:
-            reflectance_3x_calculator = Calculator(metadata['platform_name'], metadata['sensor'], metadata['name'],
-                                                   sunz_threshold=self.sun_zenith_threshold)
-        else:
-            reflectance_3x_calculator = Calculator(metadata['platform_name'], metadata['sensor'], metadata['name'])
-            self.sun_zenith_threshold = reflectance_3x_calculator.sunz_threshold
+        reflectance_3x_calculator = Calculator(metadata['platform_name'], metadata['sensor'], metadata['name'],
+                                               sunz_threshold=self.sun_zenith_threshold,
+                                               masking_limit=self.masking_limit)
```

Comment:
I'm not sure this is right. When `self.sun_zenith_threshold` is None, we want to replace it with the default value from the calculator.
This was what #1319 was about.

Reply:
Oh. That sounds wrong. I'd expect `None` to mean "do not apply SZA correction in the Calculator". I'd just fix the documentation to say `If not explicitly specified, the default value defined in Pyspectral will be used.`

But anyway, the behaviour on the Satpy side doesn't change: the default value is used if the user doesn't override it. If the user decides to override the default with `None`, they'll probably know what they are doing (which is, crash the code :grin:).

Reply:
ok, just change the `__init__` docstring then :)

Reply:
Docstring updated. Good thing, as I didn't remember to add the new kwarg there before.